sirepo/pkcli/service.py

```diff
@@ -70,38 +75,59 @@ def _handle_signals(signums):
     def _kill(*args):
         for p in processes:
             try:
+                try:
+                    for c in list(psutil.Process(p.pid).children(recursive=True)):
+                        try:
+                            c.terminate()
+                            c.wait(1)
+                        except (
+                            ProcessLookupError,
+                            ChildProcessError,
+                            psutil.NoSuchProcess,
+                        ):
+                            continue
+                        except psutil.TimeoutExpired:
+                            c.kill()
+                except Exception:
+                    pass
+
                 p.terminate()
                 p.wait(1)
             except (ProcessLookupError, ChildProcessError):
                 continue
             except subprocess.TimeoutExpired:
                 p.kill()
 
-    def _start(service):
-        c = ["pyenv", "exec", "sirepo"]
-        c.extend(service)
+    def _start(service, cwd=".", prefix=("pyenv", "exec", "sirepo")):
         processes.append(
             subprocess.Popen(
-                c,
-                cwd=str(_run_dir()),
-                env=e,
+                prefix + service,
+                cwd=str(_run_dir().join(cwd)),
+                env=env,
             )
         )
 
-    e = PKDict(os.environ)
-    e.SIREPO_JOB_DRIVER_MODULES = "local"
-    processes = []
-    with pkio.save_chdir(_run_dir()), _handle_signals((signal.SIGINT, signal.SIGTERM)):
-        try:
-            _start(["job_supervisor"])
+    try:
+        with pkio.save_chdir(_run_dir()), _handle_signals(
+            (signal.SIGINT, signal.SIGTERM)
+        ):
+            _start(("job_supervisor",))
             # Avoid race condition on creating auth db
             time.sleep(0.3)
-            _start(["service", "flask"])
+
```

Comment:
Remove extra whitespace. If you need it, it's because the code is unreadable, and should be broken into functions. Whitespace itself makes the code harder to read for newbies to the code who do not have the writer's reading style.